src/controller/CHIPDeviceController.cpp

```diff
@@ -1950,7 +1950,13 @@ void DeviceCommissioner::ParseCommissioningInfo()
         err = ParseCommissioningInfo2(info);
     }
 
-    mAttributeCache = nullptr;
+    // Move ownership of mAttributeCache to the stack, but don't release it until this function returns.
+    // This way we can don't have to make a copy while parsing commissioning info, and won't
+    // affect future commissioning steps.
+    //
+    // The stack reference needs to survive until CommissioningStageComplete and OnReadCommissioningInfo
+    // returns (this function returns).
```

Comment:
```suggestion
    // return.
```

This function returning has nothing to do with anything.  If someone wrapped a scope around this variable, they have to ensure that CommissioningStageComplete and OnReadCommissioningInfo are in that scope, but that's it.